src/claims/claims.controller.ts

```diff
@@ -0,0 +1,70 @@
+import { Body, Controller, HttpStatus, Post } from '@nestjs/common';
+import { ApiOperation, ApiTags } from '@nestjs/swagger';
+
+import { ClaimsService } from '~/claims/claims.service';
+import { ModifyClaimsDto } from '~/claims/dto/modify-claims.dto';
+import { ApiTransactionFailedResponse, ApiTransactionResponse } from '~/common/decorators/swagger';
+import { handleServiceResult, TransactionResponseModel } from '~/common/utils';
+import { PolymeshLogger } from '~/logger/polymesh-logger.service';
+
+@ApiTags('claims')
+@Controller('claims')
+export class ClaimsController {
+  constructor(
+    private readonly claimsService: ClaimsService,
+    private readonly logger: PolymeshLogger
+  ) {
+    logger.setContext(ClaimsController.name);
+  }
+
+  @ApiOperation({
+    summary: 'Get all Claims targeting an Identity',
+    description: 'This endpoint will provide a list of all the Claims made about an Identity',
+  })
+  @ApiTransactionResponse({
+    description: 'Transaction response',
+  })
+  @ApiTransactionFailedResponse({
+    [HttpStatus.UNPROCESSABLE_ENTITY]: ['Account does not have the required roles or permissions'],
+  })
+  @Post('claims/add')
+  async addClaims(@Body() args: ModifyClaimsDto): Promise<TransactionResponseModel> {
+    const serviceResult = await this.claimsService.addClaimsOnDid(args);
+
+    return handleServiceResult(serviceResult);
+  }
+
+  @ApiOperation({
+    summary: 'Edit Claims targeting an Identity',
+    description: 'This endpoint will provide a list of all the Claims made about an Identity',
+  })
+  @ApiTransactionResponse({
+    description: 'Transaction response',
+  })
+  @ApiTransactionFailedResponse({
+    [HttpStatus.UNPROCESSABLE_ENTITY]: ['Account does not have the required roles or permissions'],
+  })
+  @Post('claims/edit')
+  async editClaims(@Body() args: ModifyClaimsDto): Promise<TransactionResponseModel> {
+    const serviceResult = await this.claimsService.editClaimsOnDid(args);
+
+    return handleServiceResult(serviceResult);
+  }
+
+  @ApiOperation({
+    summary: 'Get all Claims targeting an Identity',
+    description: 'This endpoint will provide a list of all the Claims made about an Identity',
+  })
+  @ApiTransactionResponse({
+    description: 'Transaction response',
+  })
+  @ApiTransactionFailedResponse({
+    [HttpStatus.UNPROCESSABLE_ENTITY]: ['Account does not have the required roles or permissions'],
+  })
+  @Post('claims/remove')
+  async revokeClaims(@Body() args: ModifyClaimsDto): Promise<TransactionResponseModel> {
+    const serviceResult = await this.claimsService.revokeClaimsFromDid(args);
+
+    return handleServiceResult(serviceResult);
+  }
```

Comment:
Do we need a Get for claims?

There are the endpoint `@Get(':did/associated-claims')` and ` @Get(':did/issued-claims')` currently in the identities controller. I am not sure if they should get moved over to here though

Reply:
I was thinking we could move the controller method, but try to preserve the path. Maybe with taking the `target` and moving it to the path to be under identities.

Ideally we have the claims routes in one controller, and we don't make breaking changes.

If we do need a breaking change a path rename isn't that bad imo. Mark it  as such and we should get a major release.

Reply:
If we aren’t sure it will be better I think its better to err on not making the breaking change. Its not like they are terribly out of place where they are, and I don't want people to have to update their stuff if we aren't sure it will benefit us.

I think there is benefit to having a stable API, and if we are breaking things we should have a strong reason imo.

That said, as far as breaking changes go, it will be small to move a route, and I don't think many people are using this endpoint.